debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/macho/ThreadCommand.java

```diff
@@ -0,0 +1,188 @@
+ * file.
+ * 
+ * The Original Code and all software distributed under the License are
+ * distributed on an 'AS IS' basis, WITHOUT WARRANTY OF ANY KIND, EITHER
+ * EXPRESS OR IMPLIED, AND APPLE HEREBY DISCLAIMS ALL SUCH WARRANTIES,
+ * INCLUDING WITHOUT LIMITATION, ANY WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE, QUIET ENJOYMENT OR NON-INFRINGEMENT.
+ * Please see the License for the specific language governing rights and
+ * limitations under the License.
+ *******************************************************************************/
+package com.ibm.j9ddr.corereaders.macho;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.imageio.stream.ImageInputStream;
+
+public class ThreadCommand extends LoadCommand
+{
+
+	// thread flavors
+	public static final int x86_THREAD_STATE32 = 1;
+	public static final int x86_FLOAT_STATE32 = 2;
+	public static final int x86_EXCEPTION_STATE32 = 3;
+	public static final int x86_THREAD_STATE64 = 4;
+	public static final int x86_FLOAT_STATE64 = 5;
+	public static final int x86_EXCEPTION_STATE64 = 6;
+	public static final int x86_THREAD_STATE = 7;
+	public static final int x86_FLOAT_STATE = 8;
+	public static final int x86_EXCEPTION_STATE = 9;
+	public static final int x86_DEBUG_STATE32 = 10;
+	public static final int x86_DEBUG_STATE64 = 11;
+	public static final int x86_DEBUG_STATE = 12;
+	public static final int THREAD_STATE_NONE = 13;
+	public static final int x86_AVX_STATE32 = 16;
+	public static final int x86_AVX_STATE64 = (x86_AVX_STATE32 + 1);
+	public static final int x86_AVX_STATE = (x86_AVX_STATE32 + 2);
+	public static final int x86_AVX512_STATE32 = 19;
+	public static final int x86_AVX512_STATE64 = (x86_AVX512_STATE32 + 1);
+	public static final int x86_AVX512_STATE =(x86_AVX512_STATE32 + 2);
+
+	public static class ThreadState
+	{
+		int flavor;
+		int sizeInUInts;
+		byte stateBytes[];
+		Map<String, Number> registers;
+
+		public ThreadState(int flavor, int size, byte state[])
+		{
+			this.flavor = flavor;
+			this.sizeInUInts = size;
+			this.stateBytes = state;
+			registers = new TreeMap<String, Number>();
+			switch (flavor) {
+				case x86_THREAD_STATE:
+					fillX86ThreadRegisters();
+					break;
+				case x86_EXCEPTION_STATE:
+					fillX86ExceptionData();
+					break;
+				default:
+					break;
+			}
+		}
+
+		public static long readLong(byte[] data, int start)
+		{
+			return (0xFF00000000000000L & (((long) data[start + 7]) << 56))
+					| (0x00FF000000000000L & (((long) data[start + 6]) << 48))
+					| (0x0000FF0000000000L & (((long) data[start + 5]) << 40))
+					| (0x000000FF00000000L & (((long) data[start + 4]) << 32))
+					| (0x00000000FF000000L & (((long) data[start + 3]) << 24))
+					| (0x0000000000FF0000L & (((long) data[start + 2]) << 16))
+					| (0x000000000000FF00L & (((long) data[start + 1]) << 8))
+					| (0x00000000000000FFL & (data[start + 0]));
```

Comment:
Is this and similar functions below not affected by the byte-order of the core file?